### DISCUSSION TITLE
GDPR Data Export Compliance

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/cms#14330] [Ideas]
We're having a GDPR compliance issue with some European clients, where they need to ensure that we (devs) don't store / process personal data outside the EU (we're scattered around the globe). This makes it very difficult to get a fresh database backup from the production site for local development.

I'd like to urgently suggest/request a checkbox in the Utilities -> Database Backup screen called "Exclude Personal Data". When this checkbox is checked, the database dump would exclude any stored personal data (data submitted by website visitors in forms, etc.).

Importantly, 3rd-party plugins such as Freeform would need some way to inform Craft which DB tables of theirs to exclude from the database dump.

**Comment by alexanderbuergin** [comment]
Data from all user accounts have to be excluded as well. 
As well as orders in the Craft Commerce plugin

When I finish thinking about this idea, it involves a lot of data

**Reply by proimage** [reply]
Good points! I think user account data only needs to be excluded if user account creation is a public thing, right? I didn't consider that originally—in my use-case, the only user accounts belong to employees of my client, so my understanding was that those don't need to be excluded to be GDPR-compatible.
